bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/VoiceManagerImpl.java

```diff
@@ -598,13 +603,24 @@ public void listenAndAnswer(@Nullable STTService stt, @Nullable TTSService tts,
             throw new IllegalStateException(
                     "Cannot execute a simple dialog as provided locale is not supported by all services.");
         } else {
-            DialogProcessor processor = dialogProcessors.get(audioSource.getId());
-            if (processor == null) {
+            boolean isSingleDialog = false;
+            DialogProcessor activeProcessor = dialogProcessors.get(audioSource.getId());
+            cleanSingleDialogProcessors();
```

Comment:
Above (in l. 548-551) you call the clean method only if `processor == null`, here you call it always. Is there any reason to do that differently?

Reply:
I'll let both offside the if. I think in both places it will ensure you not end with a lot of finished dialog references in memory, but better to remove those as soon as posible and should not impact the performance.